operator/controllers/mlops/model_controller.go

```diff
@@ -118,8 +119,19 @@ func (r *ModelReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl
 	return ctrl.Result{}, nil
 }
 
-func (r *ModelReconciler) updateStatusFromError(ctx context.Context, logger logr.Logger, model *mlopsv1alpha1.Model, err error) {
-	model.Status.CreateAndSetCondition(mlopsv1alpha1.ModelReady, false, err.Error())
+func (r *ModelReconciler) updateStatusFromError(
+	ctx context.Context,
+	logger logr.Logger,
+	model *mlopsv1alpha1.Model,
+	canRetry bool,
+	err error,
+) {
+	modelStatus := schedulerAPI.ModelStatus_ModelFailed
+	if canRetry {
+		modelStatus = schedulerAPI.ModelStatus_ModelProgressing
```

Comment:
:information_source: The equivalent pipeline code simply says the pipeline failed if there was any issue requesting it to be loaded.  Which behaviour do we want -- indicate failure early, or be optimistic as transient glitches can happen?

Reply:
I've updated the pipeline status to not indicate failure from retryable errors: https://github.com/SeldonIO/seldon-core/pull/4834/commits/558b0a68cd26ae5f8188a08a4b8e36f9b796087c